src/test/containerTest/org/apache/cassandra/sidecar/restore/StorageClientTest.java

```diff
@@ -192,42 +210,68 @@ void testGetObject() throws Exception
     void testGetObjectHasExistingFileOnDisk() throws Exception
     {
         Path existingPath = testFolder.resolve(UUID.randomUUID().toString());
+        Files.createDirectories(existingPath);
+        Files.createFile(existingPath.resolve("key"));
         RestoreRange sliceHasFileOnDisk = getMockRange(restoreJob.jobId, testBucket, "key", checksum, existingPath);
-        File downloaded = client.downloadObjectIfAbsent(sliceHasFileOnDisk).get();
+        File downloaded = client.downloadObjectIfAbsent(sliceHasFileOnDisk, taskExecutorPool)
+                                .toCompletionStage().toCompletableFuture().get();
         assertThat(downloaded.getAbsolutePath()).isEqualTo(existingPath.resolve("key").toString());
     }
 
     @Test
     void testGetObjectThroughputRateLimited() throws Exception
     {
-        // only allow 1/4 the speed of transfer
-        client = new StorageClient(s3AsyncClient, SidecarRateLimiter.create(LARGE_FILE_IN_BYTES >> 2));
+        // only allow 1/4 the speed of transfer; each request downloads 1024 bytes (it is an extremely low value only for testing)
+        client = new StorageClient(s3AsyncClient, 1024, SidecarRateLimiter.create(LARGE_FILE_IN_BYTES >> 2));
```

Comment:
Won't this be a problem if we run tests parallelly? spot bugs is complaining about this